signal/signal.go

```diff
@@ -104,28 +105,37 @@ type Interceptor struct {
 
 	// shutdownRequestChannel is used to request the daemon to shutdown
 	// gracefully, similar to when receiving SIGINT.
-	shutdownRequestChannel chan struct{}
+	shutdownRequestChannel chan int
 
 	// quit is closed when instructing the main interrupt handler to exit.
 	// Note that to avoid losing notifications, only shutdown func may
 	// close this channel.
 	quit chan struct{}
 
+	// exitCode represents the exit code of the interceptor.
+	exitCode int
+
+	// exitCodeMutex protects the exitCode.
+	exitCodeMutex sync.Mutex
```

Comment:
would exitCode be written to or read from multiple threads?

Reply:
Not sure what threads you are referring to here but multiple subsystems send a signal to `interruptChannel` or  `shutdownRequestChannel` we want to make sure there is no racing on this.

Reply:
The comment on top of that referred to protecting the exitCode and not the channels which you are now referring to. Plus this mutex is not exported so other subsystems can't really call it. Additionally, I don't think it would result to a race condition if multiple channels send to a golang channel. What you might want to do is queue the signals for a more deterministic behaviour but that might be an overkill for this PR.

Reply:
> The comment on top of that referred to protecting the exitCode and not the channels which you are now referring to.

What I meant is When the subsystems send signals to the channels (interruptChannel and interruptChannel) this is where the code for setting the exit status code exists.

Reply:
> this is where the code for setting the exit status code exists so that's why we used mutex protected int to prevent racing on exit code.

Not an expert myself but from what you explained here and which I can see is what is happening in the code.
We do not have multiple subsystems setting the exit code. The `setExitCode` function is only called in one place the `mainInterruptHandler` goroutine. Once that is set, lnd and accompanying subsystems shutdown and so there would be no concurrency issue fetching that exitcode here: 
https://github.com/lightningnetwork/lnd/blob/f27a64ef48ac9fb5d33b15a0cd7c0990865e5914/cmd/lnd/main.go#L45

So we might not need the mutex?